src/components/Search/OpenYieldSearch.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useMemo, useRef } from 'react'
+
+import { RowFixed } from '../Row'
+import { BasicLink } from '../Link'
+
+import FormattedName from '../FormattedName'
+import { TYPE } from '../../Theme'
+
+import { useSearchData } from 'contexts/SearchData'
+import { Blue, Heading, Menu, MenuItem } from './shared'
+
+import { CG_TOKEN_API } from 'constants/index'
+
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') // $& means the whole matched string
+}
+
+const searchKeys = ['name', 'symbol']
+
+const TokenSearch = ({
+  linkPath = (token) => `/yields/token/${token.toUpperCase()}`,
+  customOnLinkClick = () => {},
+  wrapperRef,
+  value,
+  toggleMenu,
+  setValue,
+}) => {
+  const [searcheableData, setSearcheableData] = useState(useSearchData())
+  const [loading, setIsLoading] = useState(false)
+  const [dataFetched, setDataFetched] = useState(false)
+  const { protocolNames } = searcheableData
+
+  useEffect(() => {
+    const fetchProtocols = () => {
+      const urls = []
+      const maxPage = 10
+      for (let page = 1; page <= maxPage; page++) {
+        urls.push(`${CG_TOKEN_API.replace('<PLACEHOLDER>', page)}`)
+      }
+      Promise.all(urls.map((url) => fetch(url).then((resp) => resp.json()))).then((res) => {
```

Comment:
an issue here is that some queries might fail due to coingecko's rate limits, possible solutions are:
- just keep retrying failed queries (easiest)
- we maintain a complete copy of coingecko in defillama, could also pull data from that (but you'll need to make server changes)

Reply:
yeah good point. i will switch to our internal one, its `https://api.llama.fi/protocols` u are referrring to right?

Reply:
I realized that endpoint hasn't lots of base tokens, eg it has wbtc but not btc etc. Anyways I went for your first suggested option

Reply:
yeah, atm we have no endpoint that actually returns all coingecko data